build-index/tagclient/client.go

```diff
@@ -311,6 +320,33 @@ func (cc *clusterClient) do(request func(c Client) error) error {
 	return err
 }
 
+// doOnce tries the request on only one randomly chosen client without any retries if it fails.
+func (cc *clusterClient) doOnce(request func(c Client) error) error {
+	addrs := cc.hosts.Resolve().Sample(1)
+	if len(addrs) == 0 {
+		return errors.New("cluster client: no hosts could be resolved")
+	}
+	// read the only sampled addr
+	var addr string
+	for addr = range addrs {
```

Comment:
Why do we need to loop across addrs ? Can we do `addr = addrs[0]` instead ?

Reply:
`addrs` is a map, so we need to loop.